h2o-automl/src/test/java/ai/h2o/automl/preprocessing/TargetEncodingTest.java

```diff
@@ -0,0 +1,209 @@
+            Scope.track_generic(te.getTEPreprocessor());
+            assertNotNull(te.getTEModel()._parms._fold_column);
+            assertTrue(te.getTEModel()._parms._fold_column.endsWith(TargetEncoding.TE_FOLD_COLUMN_SUFFIX));
+            assertEquals(DataLeakageHandlingStrategy.KFold, te.getTEModel()._parms._data_leakage_handling);
+
+            Model.Parameters params = new DummyModel.DummyModelParameters();
+            params._train = fr._key;
+            params._nfolds = nfolds;
+            params._fold_column = null;
+
+            PreprocessingStep.Completer complete = te.apply(params);
+            assertEquals(0, params._nfolds);
+            assertNotNull(params._fold_column);
+            assertEquals(te.getTEModel()._parms._fold_column, params._fold_column);
+            assertNotEquals(fr._key, params._train);
+            Frame newTrain = params._train.get();
+            assertTrue(ArrayUtils.contains(newTrain.names(), params._fold_column));
+            assertFalse(ArrayUtils.contains(fr.names(), params._fold_column));
+            assertEquals(nfolds, newTrain.vec(params._fold_column).toCategoricalVec().cardinality());
+            complete.run();
+        } finally {
+            te.dispose();
+            Scope.exit();
+        }
+    }
+    
+    @Test
+    public void test_te_preprocessing_lifecycle_with_automl_cv_foldcolumn() {
+        aml.getBuildSpec().input_spec.fold_column = "foldc";
+        TargetEncoding te = new TargetEncoding(aml);
+        try {
+            Scope.enter();
+            te.prepare();
+            assertNotNull(te.getTEModel());
+            assertNotNull(te.getTEPreprocessor());
+            Scope.track_generic(te.getTEModel());
+            Scope.track_generic(te.getTEPreprocessor());
+            assertNotNull(te.getTEModel()._parms._fold_column);
+            assertEquals("foldc", te.getTEModel()._parms._fold_column);
+            assertEquals(DataLeakageHandlingStrategy.KFold, te.getTEModel()._parms._data_leakage_handling);
+
+            Model.Parameters params = new DummyModel.DummyModelParameters();
+            params._train = fr._key;
+            params._nfolds = 0;
+            params._fold_column = "foldc";
+
+            PreprocessingStep.Completer complete = te.apply(params);
+            assertEquals(0, params._nfolds);
+            assertNotNull(params._fold_column);
+            assertEquals("foldc", params._fold_column);
+            assertEquals(te.getTEModel()._parms._fold_column, params._fold_column);
+            assertEquals(fr._key, params._train);
+            complete.run();
+        } finally {
+            te.dispose();
+            Scope.exit();
+        }
+    }
+    
+    
+    @Test
+    public void test_automl_run_with_target_encoding_enabled() {
+        try {
+            Scope.enter();
+            AutoMLBuildSpec autoMLBuildSpec = new AutoMLBuildSpec();
+            Frame fr = parse_test_file("./smalldata/titanic/titanic_expanded.csv"); Scope.track(fr);
```

Comment:
@erin, @goldentom42 if your goal is just to try different small datasets with different thresholds, and if you have an IDE like IntelliJ, then you can just change the dataset here , run the test, it prints the leaderboard in the console...

you can also change the threshold in `TargetEncoding.java` and rerun the test.